fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluids/v1/FluidView.java

```diff
@@ -0,0 +1,121 @@
+import net.minecraft.entity.Entity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.fluids.v1.container.FluidContainer;
+import net.fabricmc.fabric.api.fluids.v1.container.SimpleFluidContainer;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.ImmutableFluidVolume;
+import net.fabricmc.fabric.api.fluids.v1.item.ItemFluidContainer;
+import net.fabricmc.fabric.api.fluids.v1.item.ItemSink;
+import net.fabricmc.fabric.api.fluids.v1.math.Drops;
+import net.fabricmc.fabric.api.fluids.v1.world.SidedFluidContainer;
+
+/**
+ * A util class for accessing fluid containers.
+ *
+ * @see Drops
+ */
+public class FluidView {
+	private FluidView() {
+	}
+
+	/**
+	 * get the fluid container for the given stack.
+	 *
+	 * @param stack the itemstack
+	 * @param sink a place for the item to put it's byproducts into, and take items when being filled
+	 * @return the fluid container for the stack
+	 */
+	public static FluidContainer getFluidContainer(ItemStack stack, ItemSink sink) {
+		Item item = stack.getItem();
+
+		if (item instanceof ItemFluidContainer) {
+			return ((ItemFluidContainer) item).getContainer(sink, stack);
+		}
+
+		return ImmutableFluidVolume.EMPTY;
+	}
+
+	/**
+	 * merge all the entity and block containers in the given location for the given face into one fluid container.
+	 *
+	 * @param world the world
+	 * @param pos the position
+	 * @param face the face
+	 * @return a fluid container for the face
+	 */
+	public static FluidContainer getFluidContainer(World world, BlockPos pos, Direction face) {
+		FluidContainer container = getBlockFluidContainer(world, pos, face);
+		Iterable<FluidContainer> entities = getEntityFluidContainersIterable(world, pos, face);
+
+		if (Iterables.isEmpty(entities)) {
+			return container;
+		}
+
+		if ((container.isImmutable() && container.isEmpty())) {
+			return new SimpleFluidContainer(entities);
+		} else {
+			return new SimpleFluidContainer(Iterables.concat(() -> Iterators.singletonIterator(container), getEntityFluidContainersIterable(world, pos, face)));
+		}
+	}
+
+	/**
+	 * get the fluid container for the block at the given location at the given side.
+	 *
+	 * @param world the world
+	 * @param pos the position
+	 * @param face the face to access
+	 * @return the fluid container
+	 */
+	public static FluidContainer getBlockFluidContainer(World world, BlockPos pos, Direction face) {
+		BlockState state = world.getBlockState(pos);
+		Block block = state.getBlock();
+
+		if (block instanceof SidedFluidContainer) {
+			return ((SidedFluidContainer) block).getContainer(world, pos, face);
+		}
+
+		return ImmutableFluidVolume.EMPTY;
+	}
+
+	private static Iterable<FluidContainer> getEntityFluidContainersIterable(World world, BlockPos pos, Direction face) {
```

Comment:
why is this Iterable but the block one is just FluidContainer

Reply:
there can be multiple entities in a block, but not multiple blocks in a block 😉

Reply:
why is this not a list again :concern:

Reply:
it's an internal method leave me alone irritatered